Backpack/Calendar/Classes/BPKCalendar.m

```diff
@@ -625,44 +682,22 @@ - (void)configureCell:(FSCalendarCell *)cell forDate:(NSDate *)date atMonthPosit
     if (monthPosition == FSCalendarMonthPositionCurrent) {
         SelectionType selectionType = SelectionTypeNone;
         RowType rowType = RowTypeMiddle;
-
-        if (!self.sameDayRange && sortedSelectedDates.count > 1 && self.selectionConfiguration.isRangeStyleSelection) {
-            NSDate *minDate = [sortedSelectedDates firstObject];
-            NSDate *maxDate = [sortedSelectedDates lastObject];
-            BOOL dateInsideRange = [BPKCalendar date:date isBetweenDate:minDate andDate:maxDate];
-            if (dateInsideRange) {
-                BOOL isMinDate = [date isEqualToDate:minDate];
-                BOOL isMaxDate = [date isEqualToDate:maxDate];
-                NSCalendar *gregorian = self.calendarView.gregorian;
-                NSDate *firstWeekday = [gregorian fs_firstDayOfWeek:date];
-                NSDate *lastWeekday = [gregorian fs_lastDayOfWeek:date];
-                NSDate *firstDayOfMonth = [gregorian fs_firstDayOfMonth:date];
-                NSDate *lastDayOfMonth = [gregorian fs_lastDayOfMonth:date];
-                BOOL isRowStart = [gregorian isDate:date inSameDayAsDate:firstWeekday] || [gregorian isDate:date inSameDayAsDate:firstDayOfMonth];
-                BOOL isRowEnd = [gregorian isDate:date inSameDayAsDate:lastWeekday] || [gregorian isDate:date inSameDayAsDate:lastDayOfMonth];
-
-                if (isRowStart && isRowEnd) {
-                    rowType = RowTypeBoth;
-                } else if (isRowStart) {
-                    rowType = RowTypeStart;
-                } else if (isRowEnd) {
-                    rowType = RowTypeEnd;
-                }
-
-                if (isMinDate) {
-                    selectionType = SelectionTypeLeadingBorder;
-                } else if (isMaxDate) {
-                    selectionType = SelectionTypeTrailingBorder;
-                } else {
-                    selectionType = SelectionTypeMiddle;
-                }
-            }
+        BOOL wholeMonthSelection = NO;
+
+        if (self.isSelectingWholeMonth) {
+            rowType = [self rowTypeForDate:date amongRangeOfDates:sortedSelectedDates];
+            selectionType = [self selectionTypeForDate:date amongRangeOfDates:sortedSelectedDates];
+            wholeMonthSelection = [self isDate:date betweenRangeOfDates:sortedSelectedDates];
+        } else if (!self.sameDayRange && sortedSelectedDates.count > 1 && self.selectionConfiguration.isRangeStyleSelection) {
+            rowType = [self rowTypeForDate:date amongRangeOfDates:sortedSelectedDates];
+            selectionType = [self selectionTypeForDate:date amongRangeOfDates:sortedSelectedDates];
         } else if ([sortedSelectedDates containsObject:date]) {
             selectionType = self.sameDayRange ? SelectionTypeSameDay : SelectionTypeSingle;
         }
 
         calendarCell.selectionType = selectionType;
         calendarCell.rowType = rowType;
+        calendarCell.wholeMonthSelection = wholeMonthSelection;
```

Comment:
Should we change this to 'isWholeMonthSelection' to indicate it is a boolean property?